example/calendar.cpp

```diff
@@ -0,0 +1,165 @@
+    auto pad_size = (week_str_size - title.size()) / 2;
+    auto aligned_title = (std::string(pad_size, ' ') + title);
+    aligned_title.resize(week_str_size, ' ');
+    return aligned_title;
+};
+
+// In: sequence<sys_day> (week days)
+// Out: std::string (week string)
+auto week_to_string = [](flux::sequence auto&& week) { 
+    bool first_week_in_month = ymd(*flux::front(week)).day() == 1d;
+    auto week_str = FLUX_FWD(week)
+                     .map(day_to_string)
+                     .flatten()
+                     .template to<std::string>();                     
+    if (week_str.size() < week_str_size) {
+        std::string padding(week_str_size - week_str.size(), ' ');
+        week_str = first_week_in_month ? padding + week_str : week_str + padding;
+    }
+
+    return week_str;
+};
+
+// In: sequence<sys_day> (month days)
+// Out: sequence<std::string> (month calendar)
+auto to_week_lines = [](flux::sequence auto&& month) { 
+    auto month_str = month_name(*flux::front(month));
+    auto week_lines = FLUX_FWD(month)
+                        .chunk_by(week_num)
+                        .map(week_to_string);
+    std::vector<std::string> pad_lines(max_weeks_in_month - flux::count(week_lines), std::string(week_str_size, ' '));
+    return flux::chain(flux::single(std::move(month_str)), 
+                       std::move(week_lines), 
+                       std::move(pad_lines),
+                       flux::single(std::string(week_str_size, row_sep)));
+};
+
+auto append_column = [](std::vector<std::string>&& months_per_line, flux::sequence auto&& month) {
+    return flux::zip(flux::from(std::move(months_per_line)), FLUX_FWD(month).template to<std::vector>())
+        .map([](const auto& v) { return std::get<0>(v) + std::get<1>(v) + col_sep; })
+        .template to<std::vector>();
+};
+
+// In: sequence<sequence<std::string>> (chunks of months per line)
+// Out: std::vector<std::string> (months organized in columns)
+auto to_columns = [](flux::sequence auto&& month_chunk) {
+    auto n_rows = month_chunk.front()->count();
+    return FLUX_FWD(month_chunk).fold(append_column, std::vector<std::string>(n_rows, col_sep));
+};
+
+const auto current_year = ymd{floor<days>(system_clock::now())}.year();
+
+struct app_args_t {
+    bool help;
+    unsigned per_line = 3;
+    ymd from = current_year / January / 1d;
+    ymd to = from + years{1};
+};
+
+[[noreturn]] void print_help_and_exit(std::string_view app_name) {
+    std::cout << "Usage: " << app_name << " [--help] [--per-line=num] [--from=year] [--to=year]\n\n"; 
+    exit(0);
+}
+
+app_args_t parse_args(int argc, char** argv) {
+    app_args_t result;
+    std::string_view app_name = argv[0];
+    std::vector<std::string> args(std::next(argv), std::next(argv, argc));
+
+    using parse_func_t = std::function<void(std::string)>;
+    std::map<std::string, parse_func_t> args_map {
+        {"--help"s, [&](std::string val) { print_help_and_exit(app_name); }},
```

Comment:
I get a very silly "unused argument" warning here

```suggestion
        {"--help"s, [&](std::string) { print_help_and_exit(app_name); }},
```

Reply:
I changed to `{"--help"s, [&]([[maybe_unused]] std::string) { print_help_and_exit(app_name); }},`. It is ok for you?

Reply:
I think just removing the name `val` is probably enough to keep the compiler happy, but `[[maybe_unused]]` doesn't do any harm